### PR TITLE
Switch from jshint to eslint

### PR DESCRIPTION
**Pull request by ahennr**
As announced in terrestris/SolrHeatmap#21 the JavaScript code linting utility was changed to [ESLint](http://eslint.org/).

Beside the `.eslintrc` and the `.eslintignore` based on those of [BasiGX](https://github.com/terrestris/BasiGX), dedicated ESLint rules for angular projects have are introduced using the [ESLint pluginfor AngularJS projects](https://www.npmjs.com/package/eslint-plugin-angular)

Please review @terrestris/owners

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+build/
+coverage/
+assets/
+node_modules/
+.grunt/
+
+Gruntfile.js
```
Comment (annarieger) [inline .eslintignore:7]: I'd prefer not to add the `Gruntfile.js` to `.gitignore` cause it contains the configuration of grunt tasks..

Just a note :)

File: app/controller/Search.js [hunk 2]
```diff
@@ -1,48 +1,54 @@
...
-        /**
-         *
-         */
-        $scope.onKeyPress = function($event) {
-            // only fire the search if Enter-key (13) is pressed
-            if (getKeyboardCodeFromEvent($event) === 13) {
-                $scope.doSearch();
+angular.module('SolrHeatmapApp')
+    .controller('SearchController', ['Map', 'HeatMapSourceGenerator', '$scope', '$window',
+        function(MapService, HeatMapSourceGeneratorService, $scope, $window) {
```
Comment (annarieger) [inline app/controller/Search.js]: `MapService` is currently not used in this class, please remove it from requires.

File: package.json [hunk 3]
```diff
@@ -27,12 +27,15 @@
...
   },
   "dependencies": {},
   "devDependencies": {
     "angular-ui-bootstrap": "^1.3.3",
     "eslint": "^2.13.0",
+    "eslint-config-angular": "^0.5.0",
+    "eslint-plugin-angular": "^1.3.0",
     "grunt": "1.0.1",
     "grunt-cli": "1.2.0",
     "grunt-contrib-clean": "^1.0.0",
```
Comment (annarieger) [inline package.json:41]: please remove `grunt-contrib-jshint` dependency from package as well (s. [here](https://github.com/ahennr/SolrHeatmap/blob/14a7c788028d48949b18aa23a692abc9d1db8839/package.json#L43))

**Comment by annarieger** [comment]
Many thanks for this great adjustments, @ahennr! :crown: I've added some very minor remarks only.

File: .eslintignore (hunk 1 above)
Comment (ahennr) [inline .eslintignore:7]: Absolutely. However I added `Gruntfile.js` this to `.eslintignore` and not to `.gitignore` :wink:. I'll keep this in here, ok?

File: package.json (hunk 3 above)
Comment (ahennr) [inline package.json:41]: Ok. Removed in 931e5a3

File: app/controller/Search.js (hunk 2 above)
Comment (ahennr) [inline app/controller/Search.js]: Alright, removed in 931e5a3

File: .eslintignore (hunk 1 above)
Comment (annarieger) [inline .eslintignore:7]: Oh, sorry for the noise, I'm blind :confused:

**Comment by ahennr** [comment]
THX for review @annarieger. Will merge.

**Pull request merged**